xcode/Subconscious/Shared/Services/AddressBookService.swift

```diff
@@ -405,21 +405,30 @@ actor AddressBookService {
         await self.addressBook.invalidateCache()
     }
     
+    private static func shouldBeUnfollowed(
+        _ entry: AddressBookEntry,
+        _ did: Did,
+        _ petname: Petname.Part?
+    ) -> Bool {
+        entry.did == did && (petname == nil || entry.identifier == petname)
+    }
```

Comment:
This addresses https://github.com/subconsciousnetwork/subconscious/issues/657 by trying to optionally refine the unfollow request using a passed petname.